From 54d2b47f8124373fe8557caef5294b225b18e074 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Sat, 3 Dec 2016 02:36:39 -0800
Subject: [PATCH 46/57] UBUNTU: SAUCE: apparmor: fix oops in bind_mnt when
 dev_path lookup fails

Bind mounts can oops when devname lookup fails because the devname is
uninitialized and used in auditing the denial.

BugLink: http://bugs.launchpad.net/bugs/1660840
Signed-off-by: John Johansen <john.johansen@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 security/apparmor/mount.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/apparmor/mount.c b/security/apparmor/mount.c
index 907d3f0a0278..e78018172eab 100644
--- a/security/apparmor/mount.c
+++ b/security/apparmor/mount.c
@@ -335,6 +335,7 @@ static int match_mnt_path_str(struct aa_profile *profile, const struct path *mnt
 		goto audit;
 	if (IS_ERR(devname)) {
 		error = PTR_ERR(devname);
+		devname = NULL;
 		info = devinfo;
 		goto audit;
 	}
-- 
2.17.1

